src/email.rs

```diff
@@ -93,7 +90,7 @@ pub fn request(app: &Config, email_addr: EmailAddress, client_id: &str, nonce: &
 pub fn verify(app: &Config, stored: &HashMap<String, String>, code: &str)
               -> BrokerResult<(String, String)> {
 
-    if code != &stored["code"] {
+    if code.to_lowercase() != (&stored["code"]).to_string() {
```

Comment:
The `to_string()` here creates a copy. Instead, if you take a reference to the `String` left-hand, that'll coerce into `&str`, and allow comparison:

`if &code.to_lowercase() != &stored["code"] { ...`